Actividad 1 — Identificación de la fuente

Propósito:reconocer la procedencia del conjunto de datos antes de utilizarlo y documentar correctamente su fuente.

Los datos utilizados en esta práctica provienen de Our World in Data (OWID)
La fuente original es **United Nations, World Population Prospects (2024)**.

Cita:
> UN, World Population Prospects (2024) – processed by Our World in Data. "Annual change in population – UN WPP" [dataset]. United Nations, "World Population Prospects"; United Nations, "World Population Prospects - Interim Update" [original data].

Preguntas

1. ¿Quién produjo originalmente los datos?
   La Organización de las Naciones Unidas (ONU), a través de su división de población, en el reporte World Population Prospects (2024).

2. ¿Qué organización los procesa para su utilización en Our World in Data?
   Our World in Data (OWID) es quien recopila, limpia y estandariza los datos originales de la ONU para publicarlos en un formato reutilizable (CSV + metadata).

3. ¿Qué representa la variable que analizarás?
   Representa el cambio anual de población (crecimiento o decrecimiento poblacional) de cada país/entidad, año por año, expresado según la unidad indicada en el archivo `.metadata.json` 

Actividad 2 — Carga y exploración de datos

Propósito:comprobar que el entorno de Python, Pandas y Jupyter funciona correctamente y realizar una primera exploración del dataset.

In [13]:
import pandas as pd
import numpy as np

In [14]:
df = pd.read_csv("../data/raw/annual-population-growth/annual-population-growth.csv")
df.head()

,Entity,Code,Year,Annual change in population,Annual population change (Projected)
0,Afghanistan,AFG,1951,103163.0,NaN
1,Afghanistan,AFG,1952,108441.0,NaN
2,Afghanistan,AFG,1953,108919.0,NaN
3,Afghanistan,AFG,1954,111251.0,NaN
4,Afghanistan,AFG,1955,119027.0,NaN


2.3 Exploración del DataFrame

In [15]:
df.head()

,Entity,Code,Year,Annual change in population,Annual population change (Projected)
0,Afghanistan,AFG,1951,103163.0,NaN
1,Afghanistan,AFG,1952,108441.0,NaN
2,Afghanistan,AFG,1953,108919.0,NaN
3,Afghanistan,AFG,1954,111251.0,NaN
4,Afghanistan,AFG,1955,119027.0,NaN


In [6]:
df.shape

(38400, 5)

In [5]:
print("Dimensiones:", df.shape)
print("\nColumnas:", df.columns.tolist())
print("\nTipos de datos:\n", df.dtypes)

Dimensiones: (38400, 5)

Columnas: ['Entity', 'Code', 'Year', 'Annual change in population', 'Annual population change (Projected)']

Tipos de datos:
 Entity                                      str
Code                                        str
Year                                      int64
Annual change in population             float64
Annual population change (Projected)    float64
dtype: object


In [16]:
df.describe()

,Year,Annual change in population,Annual population change (Projected)
count,38400.000000,1.868800e+04,1.971200e+04
mean,2025.500000,2.063019e+06,8.375481e+05
std,43.300872,9.440859e+06,5.816745e+06
min,1951.000000,-3.315935e+06,-2.291035e+07
25%,1988.000000,1.781000e+03,-9.742750e+03
50%,2025.500000,4.689800e+04,9.350000e+01
75%,2063.000000,3.021128e+05,1.411160e+05
max,2100.000000,9.337137e+07,7.206736e+07


In [7]:
print("Valores faltantes por columna:\n", df.isnull().sum())
print("\nCantidad de entidades distintas:", df["Entity"].nunique())

Valores faltantes por columna:
 Entity                                      0
Code                                     1200
Year                                        0
Annual change in population             19712
Annual population change (Projected)    18688
dtype: int64

Cantidad de entidades distintas: 256


Preguntas:
1. ¿Cuántas filas y columnas tiene el dataset? 5 columnas y 38,400 filas
2. ¿Qué tipo de datos contiene la variable Year? int64
3. ¿Existen valores faltantes? si,en code hay 1200 valores faltantes,anual change in population 19712 valores faltantes y en anual population change(projected) 18688 valores faltantes
4. ¿Cuántas entidades diferentes contiene el dataset? 256 entidades diferentes

Actividad 3 — Estadística descriptiva

Propósito: Aplicar medidas estadísticas básicas a una variable poblacional.

In [17]:
anio = 2020  
df_anio = df[df["Year"] == anio]
df_anio.head()

,Entity,Code,Year,Annual change in population,Annual population change (Projected)
69,Afghanistan,AFG,2020,1212852.0,NaN
219,Africa (UN),UN_AFR,2020,32815675.0,NaN
369,Albania,ALB,2020,-13061.0,NaN
519,Algeria,DZA,2020,747543.0,NaN
669,American Samoa,ASM,2020,-451.0,NaN


In [18]:
df_extraccion_entity = df ["Entity"]
df_extraccion_entity.head(3)

0    Afghanistan
1    Afghanistan
2    Afghanistan
Name: Entity, dtype: str

In [15]:
columna = "Annual change in population" 

media = df_anio[columna].mean()
mediana = df_anio[columna].median()
varianza = df_anio[columna].var()
desviacion = df_anio[columna].std()
minimo = df_anio[columna].min()
maximo = df_anio[columna].max()

print("Media:", media)
print("Mediana:", mediana)
print("Varianza:", varianza)
print("Desviación estándar:", desviacion)
print("Mínimo:", minimo)
print("Máximo:", maximo)

Media: 2124908.84765625
Mediana: 34293.5
Varianza: 90687335031103.64
Desviación estándar: 9522989.815761836
Mínimo: -494016.0
Máximo: 75707590.0


In [17]:
entidad_min = df_anio.loc[df_anio[columna].idxmin(), "Entity"]
entidad_max = df_anio.loc[df_anio[columna].idxmax(), "Entity"]

print("Entidad con valor mínimo:", entidad_min)
print("Entidad con valor máximo:", entidad_max)

Entidad con valor mínimo: Venezuela
Entidad con valor máximo: World


Preguntas

1. ¿Cuál es el valor de la media? es de 2,124,908.85 aproximadamente

2. ¿Cuál es el valor de la mediana? es de 34,293.5

3. ¿Cuál es el valor de la desviación estándar?
   La desviación estándar es 9,522,989.82 aproximadamente

4. ¿Qué entidad presenta el valor mínimo y cuál el máximo?
   La entidad con el valor mínimo es venezuela y la que presenta el valor máximo es World con 75,707,590

Actividad 4 — Comprobación del entorno

Propósito: comprobar que el notebook está utilizando correctamente las herramientas configuradas para la práctica.

In [18]:
import sys
print("Python:", sys.version)
print("Pandas:", pd.__version__)
print("NumPy:", np.__version__)

Python: 3.14.7 (tags/v3.14.7:823f032, Aug  5 2026, 10:51:32) [MSC v.1944 64 bit (AMD64)]
Pandas: 3.0.5
NumPy: 2.5.2


Resultados
En este notebook documente la fuente de los datos (Our World in Data, procesados a partir de United Nations, World Population Prospects 2024), se cargó y exploró el dataset de cambio anual de población, calcule los estadísticos descriptivos básicos (media, mediana, varianza, desviación estándar, mínimo y máximo) para el año seleccionado, y comprobar las versiones de Python, Pandas y NumPy utilizadas en el entorno virtual del proyecto.

In [21]:
df.iloc[:, 0:2]

,Entity,Code
0,Afghanistan,AFG
1,Afghanistan,AFG
2,Afghanistan,AFG
3,Afghanistan,AFG
4,Afghanistan,AFG
...,...,...
38395,Zimbabwe,ZWE
38396,Zimbabwe,ZWE
38397,Zimbabwe,ZWE
38398,Zimbabwe,ZWE


In [25]:
df.iloc[10000,2:3]

Year    2051
Name: 10000, dtype: object